tests/posix/c_lib_ext/src/fnmatch.c

```diff
@@ -73,10 +71,22 @@ ZTEST(posix_c_lib_ext, test_fnmatch)
 	zassert_equal(fnmatch("*/*", "a/.b", FNM_PATHNAME | FNM_PERIOD), FNM_NOMATCH);
 	zassert_ok(fnmatch("*?*/*", "a/.b", FNM_PERIOD));
 	zassert_ok(fnmatch("*[.]/b", "a./b", FNM_PATHNAME | FNM_PERIOD));
-	/* zassert_ok(fnmatch("*[[:alpha:]]/""*[[:alnum:]]", "a/b", FNM_PATHNAME)); */
+	/** Test cases for POSIX classes */
+	zassert_ok(fnmatch("*[[:alpha:]]", "Z", 0));
```

Comment:
It would be good to also verify that invalid character classes are reported as an error.

```suggestion
	rv = fnmatch("[[:foobarbaz:]]", "Z", 0);
	zassert_not_ok(rv);
	zassert_not_equal(rv, FNM_NOMATCH);
	
	zassert_ok(fnmatch("*[[:alpha:]]", "Z", 0));
```